query/physicalplan/aggregate.go

```diff
@@ -27,51 +26,56 @@ func Aggregate(
 	agg *logicalplan.Aggregation,
 	final bool,
 ) (*HashAggregate, error) {
-	var (
-		aggFunc      logicalplan.AggFunc
-		aggFuncFound bool
-
-		aggColumnExpr  logicalplan.Expr
-		aggColumnFound bool
-	)
-
-	agg.AggExpr.Accept(PreExprVisitorFunc(func(expr logicalplan.Expr) bool {
-		switch e := expr.(type) {
-		case *logicalplan.AggregationFunction:
-			aggFunc = e.Func
-			aggFuncFound = true
-		case *logicalplan.Column:
-			aggColumnExpr = e
-			aggColumnFound = true
+	aggregateColumns := make([]AggregateColumn, 0, len(agg.AggExprs))
+
+	for _, expr := range agg.AggExprs {
+		var (
+			aggColumn      AggregateColumn
+			aggFunc        logicalplan.AggFunc
+			aggFuncFound   bool
+			aggColumnFound bool
+		)
+		expr.Accept(PreExprVisitorFunc(func(expr logicalplan.Expr) bool {
```

Comment:
Maybe it's time to extract this aggColumn finding/creation logic to a separate helper function while you're here? up to you

Reply:
In what way? I think this is actually fine the way it is, but I'm up for proposals. It's definitely not the biggest offender in `aggregate.go` :grimacing: